### PR TITLE
Initial support for a working Google search API class.

### PR DESCRIPTION
**Pull request by Hedlund01**

### CODE REVIEW

File: configs/config.cfg [hunk 1]
```diff
@@ -31,27 +31,24 @@ scorer = {"@scorers":"spacy.textcat_multilabel_scorer.v2"}
 threshold = 0.5
 
 [components.textcat_multilabel.model]
-@architectures = "spacy.TextCatCNN.v1"
+@architectures = "spacy.TextCatReduce.v1"
```
Comment (Zecro-0) [inline configs/config.cfg:34]: What is the motivation for this change from CNN to Reduce?

**Review by Zecro-0 — COMMENTED** [review]
Overall looks good, but wondering a bit about the changes in the spaCy config file.

File: configs/config.cfg (hunk 1 above)
Comment (simlar-0) [reply configs/config.cfg:34]: Från SpaCys dokumentation:
> TextCatCNN has been replaced by the more general [TextCatReduce](https://spacy.io/api/architectures#TextCatReduce)

File: scraping/NLPspider/spiders/crawlingNLP.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import scrapy
```
Comment (yuvve) [inline scraping/NLPspider/spiders/crawlingNLP.py:1]: Module docstring missing

File: scripts/extract_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import json
```
Comment (yuvve) [inline scripts/extract_wrapper.py:1]: Module docstring missing

File: scripts/google_search_api.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+
```
Comment (yuvve) [inline scripts/google_search_api.py:1]: Module docstring missing

File: scripts/scraping_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
+
```
Comment (yuvve) [inline scripts/scraping_wrapper.py:1]: Module docstring missing

File: tests/test_divide_dataset.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        ]
+
+@pytest.fixture
+def temp_dataset(tmp_path):
+    """
+    Create a temporary input file 
+    The file has the same structure as the dataset.
+    """
+    input_file = tmp_path / "input_data.json"
+    with open(Path(input_file), 'w', encoding='utf-8') as f:
```
Comment (yuvve) [inline tests/test_divide_dataset.py:37]: Maybe make sure tmp_path exists before trying to open the file, so that the tests won't fail because of it:
```
Path(tmp_path).mkdir(parents=True, exist_ok=True)
```

File: tests/test_divide_dataset.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        json.dump(DATASET, f, indent=4, ensure_ascii=False)
+        return input_file
+    
+def test_main(temp_dataset, tmp_path):
+    temp_output = tmp_path / "output_data.json"
```
Comment (yuvve) [inline tests/test_divide_dataset.py:42]: Maybe make sure tmp_path exists before trying to open the file, so that the tests won't fail because of it:
```
Path(tmp_path).mkdir(parents=True, exist_ok=True)
```

File: tests/test_extract_wrapper.py
Comment (yuvve) [inline tests/test_extract_wrapper.py:1]: Looks good, just make sure paths exist before running tests:
```
Path(my_path).mkdir(parents=True, exist_ok=True)
```

File: tests/test_google_search_api.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (yuvve) [inline tests/test_google_search_api.py:1]: Module docstring missing

File: tests/test_google_wrapper.py
Comment (yuvve) [inline tests/test_google_wrapper.py:1]: - Use docstrings
- make sure paths exist: `Path(my_path).mkdir(parents=True, exist_ok=True)`

File: tests/test_label.py
Comment (yuvve) [inline tests/test_label.py:1]: Make sure paths exist before running tests:
```
Path(my_path).mkdir(parents=True, exist_ok=True)
```

**Review by yuvve — CHANGES_REQUESTED** [review]
Mostly docstrings missing and checking that paths exist before running tests.

File: tests/test_divide_dataset.py (hunk 6 above)
Comment (simlar-0) [reply tests/test_divide_dataset.py:37]: No need since the pytest fixture creates the temporary directory for us.

File: tests/test_extract_wrapper.py
Comment (simlar-0) [reply tests/test_extract_wrapper.py:1]: No need since the pytest fixture creates the temporary directory for us.

**Review by yuvve — APPROVED** [review]
